update-versions.sh

```diff
@@ -10,87 +10,95 @@
-
-    # shellcheck disable=2068 # allow the shell to split --exclude from each EXCLUDE_PACKAGES value
-    cargo set-version ${EXCLUDE_PACKAGES[@]/#/--exclude } --workspace "$version"
-}
-
-# This is a legacy holdover for updating extra toml files throughout various crates
-function update_extras() {
-    local version=$1
-
-    # ordered in a topological fashion starting from the workspace root Cargo.toml
-    # this isn't always necessary, but it's nice to use a mostly-consistent ordering
-    CARGO_TOMLS_TO_SED=(
-        ./Cargo.toml
-        ./cargo-pgx/src/templates/cargo_toml
-        ./nix/templates/default/Cargo.toml
-    )
-
-    DEPENDENCIES_TO_UPDATE=(
-        "pgx-pg-config"
-        "pgx-utils"
-        "cargo-pgx"
-        "pgx-macros"
-        "pgx-pgx-sys"
-        "pgx"
-        "pgx-tests"
-    )
-
-    SEMVER_REGEX="(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?"
-
-    for cargo_toml in ${CARGO_TOMLS_TO_SED[@]}; do
-        for dependency in ${DEPENDENCIES_TO_UPDATE[@]}; do
-            rg --passthru --no-line-number \
-            "(?P<prefix>^${dependency}.*\")(?P<pin>=?)${SEMVER_REGEX}(?P<postfix>\".*$)" \
-            -r "\${prefix}=${version}\${postfix}" \
-            ${cargo_toml} > ${cargo_toml}.tmp || true
-            mv ${cargo_toml}.tmp ${cargo_toml}
-        done
-    done
-
-}
-
-update_main_files $VERSION
-update_extras $VERSION
+if [ "$VERBOSE" == "1" ] || [ "$VERBOSE" == "true" ]; then
+  echo "Verbose output requested."
+  VERBOSE=1
+else
+  unset VERBOSE
+fi
+
+if [[ -v VERBOSE ]]; then
+  set -x
+fi
+
+# INCLUDE_FOR_DEP_UPDATES specifies an array of relative paths that point to Cargo
+# TOML files that are not automatically found by the pgx-version-updater tool but
+# still have PGX dependencies that require updating.
+INCLUDE_FOR_DEP_UPDATES=(
+  'cargo-pgx/src/templates/cargo_toml'
+  'nix/templates/default/Cargo.toml'
+)
+
+# EXCLUDE_FROM_VERSION_BUMP specifies an array of relative paths that point to
+# Cargo TOML files that should be *excluded* from package version bumps. Also used
+# below to include all pgx-example Cargo TOML files since they do not get their
+# versions bumped at release time.
+EXCLUDE_FROM_VERSION_BUMP=(
+  'cargo-pgx/src/templates/cargo_toml'
+  'nix/templates/default/Cargo.toml'
+  'pgx-version-updater/Cargo.toml'
+)
+
+# Exclude all pgx-examples Cargo.toml files from version bumping
+for file in pgx-examples/**/Cargo.toml; do
+  EXCLUDE_FROM_VERSION_BUMP+=("$file")
+done
+
+# Ensure that pgx-version-updater is installed and/or updated to the most recent version
+if command -v pgx-version-updater &> /dev/null; then
+  echo "pgx-version-updater found. Checking to see if update is necessary."
+  installed_version=$(pgx-version-updater --version | awk '{print $2}')
```

Comment:
Do we need to add a dependency on awk here?

Reply:
Unless there's a cool little Bash shortcut here to do this, the output of `pgx-version-updater --version` looks like this:

```
$ pgx-version-updater --version
pgx-version-updater 0.1.0
```

Unless there's a way to tell `clap` to only display the version and not `<program> <version>` in the output.

Reply:
```suggestion
  installed_version=$(pgx-version-updater --version)
  installed_version=${installed_version##pgx-version-updater } # strips the prefix
```